test/common/filter/config_discovery_impl_test.cc

```diff
@@ -450,10 +424,122 @@ TEST_F(FilterConfigDiscoveryImplTest, TerminalFilterInvalid) {
       EnvoyException,
       "Error: terminal filter named foo of type envoy.filters.http.router must be the last filter "
       "in a http filter chain.");
-  EXPECT_EQ(0UL,
-            scope_.counter("extension_config_discovery.http_filter.foo.config_reload").value());
+  EXPECT_EQ(0UL, scope_.counter("xds.extension_config_discovery.foo.config_reload").value());
+}
+
+// TCP listener test
+class TcpListenerFilterConfigDiscoveryImplTest
+    : public FilterConfigDiscoveryImplTest<
+          Network::ListenerFilterFactoryCb, Server::Configuration::ListenerFactoryContext,
+          TcpListenerFilterConfigProviderManagerImpl, Envoy::ProtobufWkt::Struct> {
+public:
+  const std::string getTypeUrl() const override { return "google.protobuf.Struct"; }
+  const std::string getFilterType() const override { return "listener"; }
+  const std::string getConfigReloadCounter() const override {
+    return "extension_config_discovery.tcp_listener_filter.foo.config_reload";
+  }
+  const std::string getConfigFailCounter() const override {
+    return "extension_config_discovery.tcp_listener_filter.foo.config_fail";
+  }
+};
+
+TEST_F(TcpListenerFilterConfigDiscoveryImplTest, DestroyReady) { testDestroyReady(); }
```

Comment:
IIUC at the moment the code is duplicated between Http/Tcp/Udp, and all the tests are the same.
If that's the case it might be better to make a parameterized test, where the initialization is different to each variant, and all the tests are defined once.

Reply:
+1

Reply:
ACK

Reply:
The tests are now changed to use typed test to avoid duplicated code. PTAL.